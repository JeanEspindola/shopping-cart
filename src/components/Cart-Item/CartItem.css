.item:hover {
  background-color: aliceblue;
}

.item__total {
  text-align: right;
}

.item__actions {
  text-align: right;
}