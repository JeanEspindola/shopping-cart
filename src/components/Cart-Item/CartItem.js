import React from 'react';
import PropTypes from 'prop-types';

const CartItem = () => (
  <div>
    CartItem Works!
  </div>
);

CartItem.propTypes = {
};

export default CartItem;
