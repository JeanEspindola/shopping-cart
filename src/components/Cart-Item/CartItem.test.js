import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CartItem from './CartItem';

describe('CartItem', () => {
  const item = {
    id: 1,
    product: 'product1',
    comments: 'comments1',
    price: 5.00,
    tax: 7,
    total: 5.35,
  };
  const mockDeleteItem = jest.fn();

  let Component;
  let wrapper;

  beforeEach(() => {
    Component = shallow(<CartItem
      id={item.id}
      product={item.product}
      comments={item.comments}
      price={item.price}
      tax={item.tax}
      total={item.total}
      onDeleteItem={mockDeleteItem}
    />);

    wrapper = mount(<CartItem
      id={item.id}
      product={item.product}
      comments={item.comments}
      price={item.price}
      tax={item.tax}
      total={item.total}
      onDeleteItem={mockDeleteItem}
    />);
  });

  it('renders CartItem without crashing', () => {
    expect(Component.length).toBeTruthy();
    expect(shallowToJson(Component)).toMatchSnapshot();
  });

  it('should call onDeleteItem when clicking on delete button', () => {
    const button = wrapper.find('button');
    button.simulate('click');
    expect(mockDeleteItem).toHaveBeenCalled();
  });
});
