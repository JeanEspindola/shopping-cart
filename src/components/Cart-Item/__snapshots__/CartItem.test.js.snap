// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders CartItem without crashing 1`] = `
<tr
  className="item"
>
  <td>
    1
  </td>
  <td>
    product1
  </td>
  <td>
    comments1
  </td>
  <td>
    5
    €
  </td>
  <td>
    7
    %
  </td>
  <td
    className="item__total"
  >
    5.35
    €
  </td>
  <td
    className="item__actions"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsSize="xsmall"
      bsStyle="danger"
      disabled={false}
      id="test"
      onClick={[Function]}
    >
      <Glyphicon
        bsClass="glyphicon"
        glyph="remove"
      />
    </Button>
  </td>
</tr>
`;
